coco_architecture.md

```diff
@@ -0,0 +1,177 @@
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential containers runtime on the host starts the VM TEE (The enclave).
+  - Enclave (agent) performs remote attestation: **steps 1-2 in the diagram**.
+  - Enclave (agent) gets the keys required to verify/decrypt the containers
+    image(s):  **steps 3-4 in the diagram**.
+  - Enclave (image management) downloads the container image(s) : **step 5 in
+    the diagram**.
+  - Enclave verifies/decrypts the container image(s) : **step 6 in the diagram**.
+  - Enclave starts the container workload.
+
+## Confidential Container Stack using a process-based TEE (e.g. Intel SGX)
+
+The following diagram shows how a CC solution is used to provide a confidential
+process to the end user:
+
+![CC_SGX_container](https://github.com/confidential-containers/documentation/blob/main/images/CC_SGX_container.jpg)
+
+Following is the workflow when deploying a confidential container (Kubernetes
+pod).
+We have marked in blue the changes from the confidential container using
+the VM-based TEE model:
+
+- CC workload preparation
+  - User builds the container image(s) (e.g. with `podman build`).
+  - User signs/encrypts the container image(s).
+  - User pushes the container image(s) to the image registry (e.g. with
+    `podman push`).
+- Deploying the CC workload in k8s
+  - User deploys the workload (`kubectl apply -f cc_workload.yaml`).
+  - Kubernetes schedules the workload to target host having the required
+    capability to run confidential containers.
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential containers runtime on the host starts the enclave agent.
+  - Enclave (agent) performs remote attestation: **steps 1-2 in the diagram**.
+  - Enclave (agent) gets the keys required to verify/decrypt the containers
+    image(s): **steps 3-4 in the diagram**.
+  - Enclave (image management) downloads the container image(s) : **step 5 in
+    the diagram**.
+  - Enclave verifies, decrypts and writes the container image(s) to a local
+    encrypted filesystem: **step 6 in the diagram**.
+  - The runtime starts the app enclave which reads the container bundle from the
+    encrypted filesystem.
+  - Secure use of the encrypted filesystem is facilitated by a key exchange
+    between the agent and app enclaves using either sealing or local attestation.
+
+As can be seen, the flows for process and VM-based TEEs are almost identical. It
+should be noted that process-based TEE requires a few additional software
+components like `libOS` without which the application requires re-architecting.
+This is not the case for VM-based TEEs. Conversely, Process based TEEs do not
+require a separate VM and CC-aware hypervisor.
+
+The up-to-date design for the Confidential Containers process-based architecture
+is maintained in the [enclave-cc documentation](https://github.com/confidential-containers/enclave-cc/blob/main/docs/design.md).
+
+# CNCF Confidential Containers
```

Comment:
I think this section was originally intended to be the coco specifics of generalizations made above.
That's not really the case though, as the sections above are already coco-specific. We should just move these diagrams into the section above and remove this section. The only addition in the diagrams I see is the kubelet->containerd->runtime blocks.

Alternatively the section could be demoted and called runtime and just point out that we have 2 different container runtimes.

Reply:
Yes it does feel like repetition, perhaps raising up a level the CoCo area to be more abstract? Then it makes it clearer this diagram is more about the bigger picture of what is needed to use Kubernetes - so focuses on the non CoCo pieces a bit more?

Reply:
I tried to re-arrange a little bit those 2 sections and make the cut between the two clearer, highlighting the fact that the Confidential Containers project is a concreter implementation of the "generic" architecture describe in the previous section. Please let me know if it looks better now.